src/annotator/sidebar.js

```diff
@@ -48,48 +44,58 @@ function createSidebarIframe(config) {
   return sidebarFrame;
 }
 
+const defaultConfig = {};
+
 /**
  * The `Sidebar` class creates the sidebar application iframe and its container,
  * as well as the adjacent controls.
  */
 export default class Sidebar extends Guest {
+  /**
+   * @param {HTMLElement} element
+   * @param {Record<string, any>} config
+   */
   constructor(element, config) {
-    if (config.theme === 'clean' || config.externalContainerSelector) {
-      delete config.pluginClasses.BucketBar;
-    }
-
-    let externalContainer = null;
-
-    if (config.externalContainerSelector) {
-      externalContainer = document.querySelector(
-        config.externalContainerSelector
-      );
-    }
+    super(element, { ...defaultConfig, ...config });
 
     let externalFrame;
     let frame;
 
-    if (externalContainer) {
-      externalFrame = externalContainer;
+    if (config.externalContainerSelector) {
+      externalFrame =
+        document.querySelector(config.externalContainerSelector) || element;
     } else {
       frame = document.createElement('div');
       frame.style.display = 'none';
       frame.className = 'annotator-frame annotator-outer';
 
       if (config.theme === 'clean') {
         frame.classList.add('annotator-frame--theme-clean');
+      } else {
+        // The instantiation of the BucketBar plugin can't be done as other plugin.
+        // The sidebar must be inserted inside a shadow DOM which is not accessible
+        // via a common querySelector.
+        this.plugins.BucketBar = new BucketBar(frame, {}, this);
       }
 
-      element.appendChild(frame);
+      // Undocumented switch to enable/disable the wrapping of the sidebar inside a shadow DOM
```

Comment:
Do you have a use case for this switch or is it effectively a "feature flag" that we can deploy quickly if we find that the change causes problems? If it is the latter, I suggest we schedule a task to remove it in the next sprint, once the change has been deployed for a week or two and we're happy that there are no major issues.

Reply:
It is a `feature flag`. Yes, I agree it should be removed once we are happy.

Reply:
OK. Can you add a note here that we should remove this flag once the change has been deployed for a week or two?

Reply:
👍🏼